crates/biome_service/src/workspace.rs

```diff
@@ -919,161 +943,223 @@ impl From<&str> for PatternId {
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase")]
 pub struct IsPathIgnoredParams {
-    pub biome_path: BiomePath,
+    pub project_key: ProjectKey,
+    pub path: BiomePath,
     pub features: FeatureName,
 }
 
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase")]
-pub struct RegisterProjectFolderParams {
+pub struct OpenProjectParams {
+    /// The path to open
+    pub path: BiomePath,
+
+    /// Whether the folder should be opened as a project, even if no
+    /// `biome.json` can be found.
+    pub open_uninitialized: bool,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase")]
+pub struct ScanProjectFolderParams {
+    pub project_key: ProjectKey,
+
+    /// Optional path within the project to scan.
+    ///
+    /// If omitted, the project is scanned from its root folder.
+    ///
+    /// This is a potential optimization that allows scanning to be limited to
+    /// a subset of the full project. Clients should specify it to indicate
+    /// which part of the project they are interested in. The server may or may
+    /// not use this to avoid scanning parts that are irrelevant to clients.
     pub path: Option<BiomePath>,
-    pub set_as_current_workspace: bool,
 }
 
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase")]
-pub struct UnregisterProjectFolderParams {
-    pub path: BiomePath,
+pub struct CloseProjectParams {
+    pub project_key: ProjectKey,
 }
 
 pub trait Workspace: Send + Sync + RefUnwindSafe {
-    /// Returns the filesystem implementation to open files with.
-    fn fs(&self) -> &dyn FileSystem;
+    // --- PROJECT-LEVEL METHODS ---
 
-    /// Checks whether a certain feature is supported. There are different conditions:
-    /// - Biome doesn't recognize a file, so it can't provide the feature;
-    /// - the feature is disabled inside the configuration;
-    /// - the file is ignored
-    fn file_features(
-        &self,
-        params: SupportsFeatureParams,
-    ) -> Result<FileFeaturesResult, WorkspaceError>;
+    /// Opens a project within the workspace.
```

Comment:
This file contains the most interesting changes. I hope the docs here help describe my intention as to where this is going.